src/Symfony/Component/Console/Application.php

```diff
@@ -390,9 +390,9 @@ public function register($name)
      *
      * If a Command is not enabled it will not be added.
      *
-     * @param Command[] $commands An array of commands
+     * @param Command[] $commands An array or iterator of commands
      */
-    public function addCommands(array $commands)
+    public function addCommands(iterable $commands)
```

Comment:
That would be a BC break unfortunately, as it's a public method, so we can't change this prototype before 5.0.

Reply:
it's not a BC break when PHP 7.1 is required in Symfony 4, which is still beta. It's a super class/primative type hint, which accepts arrays as well as iterators. It only allows more than before, however though in the event this method is overriden in a sub-class, then it'd not satisfy the function signature, if that counts.

Reply:
Application API expects `$commands` to be an array elsewhere, passing an iterable here would break at e.g. `find()`. So we would need to `iterator_to_array()`. Is there a real benefit? I don't see one personally.

Reply:
Oh, we iterate over the result just below, nevermind.
Anyway @sroze is right, any subclass overriding this method and accessing the argument as an array would break.